tuf/client_rework/updater_rework.py

```diff
@@ -147,22 +147,30 @@ def download_target(self, target: Dict, destination_directory: str):
         This method performs the actual download of the specified target.
         The file is saved to the 'destination_directory' argument.
         """
-        try:
-            for temp_obj in mirrors.mirror_target_download(
-                target, self._mirrors, self._fetcher
-            ):
 
-                self._verify_target_file(temp_obj, target)
-                # break? should we break after first successful download?
+        temp_obj = None
+        file_mirrors = mirrors.get_list_of_mirrors(
+            "target", target["filepath"], self._mirrors
+        )
 
-                filepath = os.path.join(
-                    destination_directory, target["filepath"]
+        for file_mirror in file_mirrors:
+            try:
+                temp_obj = download.download_file(
+                    file_mirror, target["fileinfo"]["length"], self._fetcher
                 )
-                sslib_util.persist_temp_file(temp_obj, filepath)
-        # pylint: disable=try-except-raise
-        except Exception:
-            # TODO: do something with exceptions
-            raise
+                _check_file_length(temp_obj, target["fileinfo"]["length"])
+                temp_obj.seek(0)
+                _check_hashes_obj(temp_obj, target["fileinfo"]["hashes"])
+                break
+
+            except Exception as exception:
+                if temp_obj:
+                    temp_obj.close()
```

Comment:
You're right, thanks for noticing. I added `temp_obj = None` for now, since changing the download function would mean a difference between metadata and targets. I'll keep this in mind.